shared/actions/fs/platform-specific.desktop.js

```diff
@@ -181,33 +180,33 @@ function* installFuseSaga(): Saga.SagaGenerator<any, any> {
   yield Saga.put(FsGen.createInstallFuseResult({kextPermissionError}))
   yield Saga.put(FsGen.createFuseStatus())
   yield Saga.put(FsGen.createSetFlags({fuseInstalling: false}))
+  // TODO: do something like uninstallConfirmSaga here
 }
 
-function uninstallKBFSConfirmSaga(action: FsGen.UninstallKBFSConfirmPayload) {
-  SafeElectron.getDialog().showMessageBox(
+const uninstallKBFSConfirm = (action: FsGen.UninstallKBFSConfirmPayload) =>
+  new Promise((resolve, reject) => SafeElectron.getDialog().showMessageBox(
     null,
     {
       buttons: ['Remove & Restart', 'Cancel'],
       detail: `Are you sure you want to remove Keybase from ${fileUIName} and restart the app?`,
       message: `Remove Keybase from ${fileUIName}`,
       type: 'question',
     },
-    resp => (resp ? undefined : action.payload.onSuccess())
+    resp => resolve(resp))
   )
-}
-
-function uninstallKBFS() {
-  return Saga.call(RPCTypes.installUninstallKBFSRpcPromise)
-}
 
-function uninstallKBFSSuccess(result: RPCTypes.UninstallResult) {
-  // Restart since we had to uninstall KBFS and it's needed by the service (for chat)
-  SafeElectron.getApp().relaunch()
-  SafeElectron.getApp().exit(0)
-}
+const uninstallKBFSConfirmSuccess = resp =>
+  (resp ? undefined : Saga.sequentially([
+    Saga.call(RPCTypes.installUninstallKBFSRpcPromise),
+    Saga.call(() => new Promise((resolve, reject) => {
+        // Restart since we had to uninstall KBFS and it's needed by the service (for chat)
+        SafeElectron.getApp().relaunch()
+        SafeElectron.getApp().exit(0)
+      }
```

Comment:
Do we need to resolve here or does it not matter since we are restarting the process?

Reply:
I think it doesn't matter because it's restarting.